### PR TITLE
Lesson 05-07: Character device driver

### PR DESCRIPTION
**Pull request by JohnTheUnigoat**
Created character device driver, /proc and /sys interfaces, testing bash script and testing instructions.

### CODE REVIEW

File: 02-KernelModules_CharacterDevice/03/MyModuleTest/char_driver.c [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+	}
+
+	buffer = kzalloc(alloc_size, GFP_KERNEL);
+
+	alloc_chrdev_region(&dev_num, 0, NUMBER_OF_DEVICES, DEVICE_NAME);
+	chdev_class = class_create(THIS_MODULE, CLASS_NAME);
+	chdev = device_create(chdev_class, NULL, dev_num, NULL, DEVICE_NAME);
+
+	cdev_init(&dev_struct, &fops);
+	cdev_add(&dev_struct, dev_num, NUMBER_OF_DEVICES);
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/03/MyModuleTest/char_driver.c]: cdev_add() can return an error. Plz handle it.

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
It's almost perfect.

File: 02-KernelModules_CharacterDevice/03/MyModuleTest/char_driver.c (hunk 1 above)
Comment (JohnTheUnigoat) [reply 02-KernelModules_CharacterDevice/03/MyModuleTest/char_driver.c]: Done.

**Comment by galkinletter** [comment]
Ok

**Pull request merged**